src/commands/Information/userinfo.ts

```diff
@@ -122,114 +121,59 @@ export default class extends Command {
 				.setAuthor(`${user.username}#${user.discriminator}`, user.avatarUrl)
 				.setDescription(description)
 				.setFooter(
-					await this.client.bulbutils.translate("global_executed_by", message.guild?.id, {
-						user: message.author,
+					await this.client.bulbutils.translate("global_executed_by", context.guild?.id, {
+						user: context.author,
 					}),
-					<string>message.author.avatarURL({ dynamic: true }),
+					<string>context.author.avatarURL({ dynamic: true }),
 				)
 				.setTimestamp();
 
-			const msg = await message.channel.send({ embeds: [embed], components });
+			const msg = await context.channel.send({ embeds: [embed], components });
 
-			const filter = (i: any) => i.user.id === message.author.id;
+			const filter = (i: any) => !i.bot;
 			const collector = msg.createMessageComponentCollector({ filter, time: 15000 });
 
-			collector.on("collect", async (interaction: ButtonInteraction) => {
-				if (interaction.customId === "warn") {
-					msg.edit({ components: [rowDisabled] });
-					await interaction.reply({
-						content: await this.client.bulbutils.translate("userinfo_interaction_confirm", message.guild?.id, {
-							action: await this.client.bulbutils.translate("mod_action_types.warn", message.guild?.id, {}),
-							target: user,
-						}),
-					});
-
-					const filter = m => m.content && m.author.id === message.author.id;
-					const collector = interaction.channel?.createMessageCollector({ filter, time: 15000, max: 1 });
-
-					collector?.on("collect", async m => {
-						let cArgs: string[] = [user.id, ...m.content.split(/ +/g)];
-
-						const command = Command.resolve(this.client, "warn")!;
-						const reason = await command.validate(message, cArgs);
-						if (reason !== undefined) {
-							if (reason) {
-								await message.channel.send(reason);
-							}
-						} else {
-							await command.run(message, cArgs);
-						}
-						await m.delete();
-					});
-
-					collector?.on("end", async () => {
-						await interaction.deleteReply();
-					});
+			collector.on("collect", async (interaction: ButtonInteraction): Promise<void> => {
+				if (interaction.user.id !== context.author.id)
+					return void (await interaction.reply({
+						content: await this.client.bulbutils.translate("global_missing_permissions", context.guildId!, {}),
+						ephemeral: true
+					}));
+
+				const command = Command.resolve(this.client, interaction.customId);
+				if (!command)
+					return void (await interaction.reply({
+						content: await this.client.bulbutils.translate("global_error.unknown", context.guildId!, {}),
+						ephemeral: true
+					}));
+
+				const reason = await command.validate(await getCommandContext(interaction), [user.id, ""]);
+				if (reason !== undefined) {
+					if (reason)
+						await interaction.reply({content: reason, ephemeral: true});
+					return;
```

Comment:
Wanted to call attention to this refactor to simplify the code for the `userinfo` action buttons